docs/docs/how_to/callbacks_custom_events.ipynb

```diff
@@ -0,0 +1,360 @@
+    "in a custom callback handler or via the [Astream Events API](/docs/concepts/#astream_events).\n",
+    "\n",
+    "For example, if you have a long running tool with multiple steps, you can dispatch custom events between the steps and use these custom events to monitor progress.\n",
+    "You could also surface these custom events to an end user of your application to show them how the current task is progressing.\n",
+    "\n",
+    "To dispatch a custom event you need to decide on two attributes for the event: the `name` and the `data`.\n",
+    "\n",
+    "| Attribute | Type | Description                                                                                              |\n",
+    "|-----------|------|----------------------------------------------------------------------------------------------------------|\n",
+    "| name      | str  | A user defined name for the event.                                                                       |\n",
+    "| data      | Any  | The data associated with the event. This can be anything, though we suggest making it JSON serializable. |\n",
+    "\n",
+    "\n",
+    ":::{.callout-important}\n",
+    "* Dispatching custom callback events requires `langchain-core>=0.2.15`.\n",
+    "* Custom callback events can only be dispatched from within an existing `Runnable`.\n",
+    "* If using `astream_events`, you must use `version='v2'` to see custom events.\n",
+    "* Sending or rendering custom callbacks events in LangSmith is not yet supported.\n",
+    "* If using python<=3.10 and async, you must propagate the `RunnableConfig` object manually (see example below).\n",
+    ":::"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "\n",
+      "\u001b[1m[\u001b[0m\u001b[34;49mnotice\u001b[0m\u001b[1;39;49m]\u001b[0m\u001b[39;49m A new release of pip is available: \u001b[0m\u001b[31;49m24.0\u001b[0m\u001b[39;49m -> \u001b[0m\u001b[32;49m24.1.2\u001b[0m\n",
+      "\u001b[1m[\u001b[0m\u001b[34;49mnotice\u001b[0m\u001b[1;39;49m]\u001b[0m\u001b[39;49m To update, run: \u001b[0m\u001b[32;49mpython -m pip install --upgrade pip\u001b[0m\n",
+      "Note: you may need to restart the kernel to use updated packages.\n"
+     ]
+    }
+   ],
+   "source": [
+    "# | output: false\n",
+    "# | echo: false\n",
+    "\n",
+    "%pip install -qU langchain-core"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Astream Events API\n",
+    "\n",
+    "The most useful way to consume custom events is via the [Astream Events API](/docs/concepts/#astream_events).\n",
+    "\n",
+    "We can use the `async` `adispatch_custom_event` API to emit custom events in an async setting. \n",
+    "\n",
+    ":::caution COMPATIBILITY\n",
```

Comment:
This is a bit redundant with the above callout?

Reply:
Will move to the top